### PR TITLE
mobile/python: Add `requirements.in` and update

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
im mostly avoiding landing mobile/docs dependabot updates as i cant test the docs output easily

this updates the deps in one go - i can check the render once it lands and revert if necessary

also adds `requirements.in` to improve dep handling, pegged at current version resolution

File: mobile/docs/requirements.txt [hunk 1]
```diff
@@ -1,202 +1,289 @@
...
-    --hash=sha256:3cfc1c6756ef1b132687b813ec6ea2214cb7a7e5d1dcb2772006cb895a0fa469
-sphinx-rtd-theme==1.2.0 \
-    --hash=sha256:a0d8bd1a2ed52e0b338cbe19c4b2eef3c5e7a048769753dac6a9f059c7b641b8 \
-    --hash=sha256:f823f7e71890abe0ac6aaa6013361ea2696fc8d3e1fa798f463e82bdb77eeff2
+    --hash=sha256:09b16deb8547d3412ad7b590689584cd0fe25ec8db3be37788be3810cbf19cb1 \
+    --hash=sha256:c8e1716e83cc398ae16824e5572ae04e0d9fc2c6b985fb0f900f5f0c96ecba1a
+    # via
+    #   -r requirements.in
+    #   sphinx
+sphinx==6.2.1 \
```
Comment (phlax) [inline mobile/docs/requirements.txt:221]: argh - this is downgrading sphinx - but it was hitting dependency resolution problems - ill try and update

Comment (phlax) [reply mobile/docs/requirements.txt]: i remember now - we fork sphinx-rtd-theme as its ~abandonware/unmaintained

i have added similar hacks to ours which hopefully will work

Comment (phlax) [reply mobile/docs/requirements.txt]: seems to ...

image

**Comment by phlax** [comment]
posting again for posterity

image

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Comment by phlax** [comment]
pushing through as relevant parts of ci are passing and pointless wasting full ci run for unrelated changes

**Pull request merged**